app/models/miq_ae_namespace.rb

```diff
@@ -106,11 +110,7 @@ def domain_name
   end
 
   def domain
-    if domain?
-      self
-    elsif (ns = ancestors.last) && ns.domain?
-      ns
-    end
+    root
```

Comment:
Wonder if this should just be `alias domain root`

Reply:
I realized this may not be right because of the `$` domain check...This might be better...

```ruby
root if root.domain?
```